### PR TITLE
Remove trailing `-` character for OS value in the list of agents

### PR DESCRIPTION
**Pull request by asteriscos**
### Description
Team,
this PR removes the trailing `-` character for OS version is empty value in the list of agents
 
### Issues Resolved
Closes https://github.com/wazuh/wazuh-kibana-app/issues/4809

### Evidence
image

### Test

- Register 2 or more agents in the manager
- Create at least 2 groups
- Add 1 on the agents to both groups and remove it from the default group
- Register another agents from _Wazuh Dev tools_ using this request
```
POST /agents
{
  "name": "NewAgent"
}
```

- Go to the Agents table
- Verify empty fields contain a `-` value
- Verify the OS column has a `-` only if it's empty or as a separator between the OS name and version
- Verify that if the OS has only the name, but has no version then it doesn't include a `-` separator

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by Tostti — DISMISSED** [review]
LGTM

**Review by yenienserrano — DISMISSED** [review]
LGTM

**Review by AlexRuiz7 — APPROVED** [review]
LGTM

**Review by yenienserrano — DISMISSED** [review]

**Review by Tostti — APPROVED** [review]
LGTM

**Review by AlexRuiz7 — DISMISSED** [review]

File: public/controllers/agent/components/agents-table.js [hunk 1]
```diff
@@ -291,7 +288,7 @@ export const AgentsTable = withErrorBoundary(
         ip: agent.ip,
         status: agent.status,
         group_config_status: agent.group_config_status,
-        group: checkField(agent.group),
+        group: agent?.group || [],
```
Comment (Desvelao) [inline public/controllers/agent/components/agents-table.js:291]: **thought**: is this change breaking the render of the group? 

https://github.com/wazuh/wazuh-kibana-app/blob/fe811908c683a25c85ad0acf891766f6c26bf2b2/public/controllers/agent/components/agents-table.js#L481

We should review the cases and decide what displays when there is not a group. If this change doesn't break the funcitonality, then we could remove the check of if the group is `-`.

**Review by AlexRuiz7 — APPROVED** [review]

File: public/controllers/agent/components/agents-table.js (hunk 1 above)
Comment (asteriscos) [reply public/controllers/agent/components/agents-table.js]: Good catch! Fixed it in https://github.com/wazuh/wazuh-kibana-app/pull/4828/commits/6e47fe8914d1f79f30d90876065ba8dd58714f17

Screenshot from 2023-01-19 12-20-53

**Review by yenienserrano — DISMISSED** [review]

**Review by AlexRuiz7 — APPROVED** [review]

**Comment by Desvelao** [comment]
# Kibana 7.10.2

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox |
| --- |  --- | --- |
| Verify empty fields contain a - value | :green_circle: | :green_circle: |
| Verify the OS column has a - only if it's empty or as a separator between the OS name and version | :green_circle: | :green_circle: |
| Verify that if the OS has only the name, but has no version then it doesn't include a - separator | :green_circle: | :green_circle: |

**Details**
<details>
<summary>:green_circle: Verify empty fields contain a - value</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

</details>

<details>
<summary>:green_circle: Verify the OS column has a - only if it's empty or as a separator between the OS name and version</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

</details>

<details>
<summary>:green_circle: Verify that if the OS has only the name, but has no version then it doesn't include a - separator</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

</details>

**Review by Desvelao — APPROVED** [review]
**review**

:green_circle: code
:green_circle: test

**Review by AlexRuiz7 — APPROVED** [review]

**Pull request merged**